compiler/luci/pass/src/SubstituteExpandDimsToReshapePass.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/SubstituteExpandDimsToReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+#include <bitset>
+#include <vector>
+
+/**
+ * @brief Convert expand_dims op to reshape op
+ * @example
+ *             input.shape = [2,3,4]
+ *             expand_dims(input, axis=1)
+ *
+ *             can be converted to
+ *
+ *             reshape(input, [2,1,3,4])
+ */
+namespace
+{
+
+int32_t unknown_dim_count(luci::CircleNode *node)
+{
+  int32_t count = 0;
+
+  for (uint32_t i = 0; i < node->rank(); ++i)
+    if (!node->dim(i).known())
+      ++count;
+
+  return count;
+}
+
+bool substitute_expand_dims_to_reshape(luci::CircleNode *node)
+{
+  auto target_node = dynamic_cast<luci::CircleExpandDims *>(node);
+  if (target_node == nullptr)
+    return false;
+  if (target_node->shape_status() != luci::ShapeStatus::VALID) //
+    return false;
+  auto input_node = loco::must_cast<luci::CircleNode *>(target_node->input());
+  if (input_node->rank() <= 0)
+    return false;
+  if (input_node->shape_status() != luci::ShapeStatus::VALID) //
+    return false;
+  auto axis_node = loco::must_cast<luci::CircleConst *>(target_node->axis());
```

Comment:
There is no restriction that axis should be a constant.

```suggestion
  auto axis_node = dynamic_cast<luci::CircleConst *>(target_node->axis());
```

Reply:
✔️ marked!